pkg/tempdir/tempdir.go

```diff
@@ -0,0 +1,126 @@
+package tempdir
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/containers/storage/pkg/lockfile"
+	"github.com/sirupsen/logrus"
+)
+
+// TODO: implement alternative for: system.EnsureRemoveAll(dir)
+
+// TempDir represents a temporary directory that is created in a specified root directory.
+type TempDir struct {
+	rootDir      string
+	tempDirPath  string
+	lockFilePath string
+	lock         *lockfile.LockFile
```

Comment:
(Eventually, the semantics of `lock` will need to be precisely documented. That can wait until the last phases of the prototype.)